### PR TITLE
Change Data Storage schema

### LINKED ISSUE #41: Change schema to conform with CCAddress specs
Remove `address` field, rename `tags` filed to `roles`, and add `tele` in JSON form

### PR DESCRIPTION
**Pull request by Rachelcoll**
Fixed #41 

### Changes: 
- Rename `JsonAdaptedTag` to `JsonAdapted Role`
- Rename `tags` in `JsonAdaptedPerson` to `roles`
- Add `tele` field (String) in `JsonAdaptedPerson`
- Remove `address` field from `JsonAdaptedPerson`

### Things to do in other parts:
- Add `tele` (type: String) field to `Person` and update `Person` constructor
- Create `Tele` type
- **Rename `Tag` to `Role` (currently not changing cause not sure if it will affect other parts)**

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedTag.java
Comment (shizy) [inline src/main/java/seedu/address/storage/JsonAdaptedTag.java:1]: Might want to change the name of the file also

**Review by shizy — APPROVED** [review]
Overall LGTM! We probably also want to change the JsonAdaptedTag.java file name later on

File: src/main/java/seedu/address/storage/JsonAdaptedTag.java
Comment (Rachelcoll) [reply src/main/java/seedu/address/storage/JsonAdaptedTag.java:1]: Thanks! already changed

**Review by superb-sushi — APPROVED** [review]
LGTM

**Pull request merged**